src/guiguts/file.py

```diff
@@ -747,6 +749,46 @@ def update_page_marks_from_flags(self) -> bool:
         maintext().set_modified(True)
         return True
 
+    def update_page_marks_from_ppgen(self) -> bool:
+        """If `.bn` ppgen commands in file, replace all existing page marks.
+
+        Command is of the form: `.bn a001.png // optional comment`
+        We only want `a001`
+
+        Returns:
+            True if `.bn` commands were present.
+        """
+        regex = r"^ *\.bn *(.+?)\.png"
```

Comment:
ppgen dot-commands **must** have the dot as the first character of the line. So the leading ` *` right after the caret should be removed.

After `bn` is ` *` for "zero or more spaces". Here, zero spaces would be invalid, so this should become ` +?`.

New regex: `^\.bn +?(.+?)\.png`

Reply:
Ah, I didn't know about the first point, but I did know about the second - thanks for catching it!

Reply:
Fixed